### DISCUSSION TITLE
v2.0.0-rc0001

### DISCUSSION THREAD

**Discussion by augustoproiete** [cake-build/cake#3680] [Announcements]
As part of this release we had [65 issues](https://github.com/cake-build/cake/milestone/71?closed=1) closed.

__Breaking changes__

- [__#3654__](https://github.com/cake-build/cake/issues/3654) IsRunningOnAzurePipelines should ignore agent type (@pascalberger)
- [__#3631__](https://github.com/cake-build/cake/issues/3631) Refactor GitHub Actions Paths (@devlead)
- [__#3610__](https://github.com/cake-build/cake/issues/3610) Remove TFBuildProvider (@pascalberger)
- [__#3590__](https://github.com/cake-build/cake/issues/3590) Directories in AzurePipelinesBuildInfo are FilePaths - FilePath.GetDirectory then inconsistent (@devlead)
- [__#3581__](https://github.com/cake-build/cake/issues/3581) Stop shipping Cake.Portable Chocolatey package and Cake Homebrew formulae (@devlead)
- [__#3579__](https://github.com/cake-build/cake/issues/3579) Stop shipping Cake runner for .NET Framework and Cake runner for .NET Core (@devlead)
- [__#3577__](https://github.com/cake-build/cake/issues/3577) Remove ReverseDependencyAttribute (@pascalberger)
- [__#3572__](https://github.com/cake-build/cake/issues/3572) Only build for TargetFrameworks netcoreapp3.1, net5.0 and net6.0 (@devlead)
- [__#3282__](https://github.com/cake-build/cake/issues/3282) GitVersion Tool: Rename verbosity values to match GitVersion values (@joshuacookdev)
- [__#3222__](https://github.com/cake-build/cake/issues/3222) Add Xamarin.iOS platform targets to MSBuildSettings PlatformTarget enumeration (@Marusyk)
- [__#3151__](https://github.com/cake-build/cake/issues/3151) Add support for Engine event hooks after execution as well as before (@BlythMeister)
- [__#3003__](https://github.com/cake-build/cake/issues/3003) Remove DependencyAttribute (@pascalberger)
- [__#2872__](https://github.com/cake-build/cake/issues/2872) Bump eol target frameworks (@gitfool)
- [__#2788__](https://github.com/cake-build/cake/issues/2788) Tool:OpenCover - the register-setting should be an option, rather than a string (@nils-a)
- [__#1111__](https://github.com/cake-build/cake/issues/1111) DotNetCoreRestore: dotnet restore no longer supports globbing (@augustoproiete, @RLittlesII)

__New Features__

- [__#3341__](https://github.com/cake-build/cake/issues/3341) Introduce DotNet aliases (synonyms to DotNetCore aliases (@augustoproiete)
- [__#3627__](https://github.com/cake-build/cake/issues/3627) Add GitHub Actions AddPath Command (@devlead)
- [__#3628__](https://github.com/cake-build/cake/issues/3628) Add GitHub Actions SetEnvironmentVariable Command (@devlead)
- [__#3629__](https://github.com/cake-build/cake/issues/3629) Add GitHub Actions UploadArtifact Command (@devlead)
- [__#3630__](https://github.com/cake-build/cake/issues/3630) Add GitHub Actions Environment properties (@devlead)

__Improvements__

- [__#3662__](https://github.com/cake-build/cake/issues/3662) Update Microsoft.CodeAnalysis.CSharp.Scripting to 4.0.0-6.final (@devlead)
- [__#3647__](https://github.com/cake-build/cake/issues/3647) Display message of criteria when task fails to run due to criteria not being met (@augustoproiete)
- [__#3644__](https://github.com/cake-build/cake/issues/3644) Add DotNetNuGetUpdateSource aliases (synonym to DotNetCoreNuGetUpdateSource) (@augustoproiete)
- [__#3643__](https://github.com/cake-build/cake/issues/3643) Add DotNetNuGetRemoveSource aliases (synonym to DotNetCoreNuGetRemoveSource) (@augustoproiete)
- [__#3642__](https://github.com/cake-build/cake/issues/3642) Add DotNetNuGetListSourceSettings (derived from to DotNetNuGetSource) (@augustoproiete)
- [__#3641__](https://github.com/cake-build/cake/issues/3641) Add DotNetNuGetHasSource aliases (synonym to DotNetCoreNuGetHasSource) (@augustoproiete)
- [__#3640__](https://github.com/cake-build/cake/issues/3640) Add DotNetNuGetEnableSource aliases (synonym to DotNetCoreNuGetEnableSource) (@augustoproiete)
- [__#3639__](https://github.com/cake-build/cake/issues/3639) Add DotNetNuGetDisableSource aliases (synonym to DotNetCoreNuGetDisableSource) (@augustoproiete)
- [__#3607__](https://github.com/cake-build/cake/issues/3639) Add EnableCompressionInSingleFile to DotNetCorePublishSettings (@SIkebe)
- [__#3599__](https://github.com/cake-build/cake/issues/3599) Add VS2022 to default MSBuild Resolver (@devlead)
- [__#3598__](https://github.com/cake-build/cake/issues/3598) Remove Preview from VS2022 MSBuild Resolver (@devlead)
- [__#3595__](https://github.com/cake-build/cake/issues/3595) Update Autofac to 6.3.0 (@devlead)
- [__#3593__](https://github.com/cake-build/cake/issues/3593) Update Microsoft.CodeAnalysis.CSharp.Scripting to 4.0.0-5.final (@devlead)
- [__#3591__](https://github.com/cake-build/cake/issues/3591) Update Microsoft.NETCore.Platforms to 6.0.0-rc.2.21480.5 (@devlead)
- [__#3555__](https://github.com/cake-build/cake/issues/3555) Add DotNetNuGetAddSource aliases (synonym to DotNetCoreNuGetAddSource) (@augustoproiete)
- [__#3554__](https://github.com/cake-build/cake/issues/3554) Add DotNetNuGetDelete aliases (synonym to DotNetCoreNuGetDelete) (@augustoproiete)
- [__#3553__](https://github.com/cake-build/cake/issues/3553) Add DotNetNuGetPush aliases (synonym to DotNetCoreNuGetPush) (@augustoproiete)
- [__#3552__](https://github.com/cake-build/cake/issues/3552) Add DotNetPack alias (synonym to DotNetCorePack) (@augustoproiete)
- [__#3551__](https://github.com/cake-build/cake/issues/3551) Add DotNetPublish alias (synonym to DotNetCorePublish) (@augustoproiete)
- [__#3550__](https://github.com/cake-build/cake/issues/3550) Add DotNetVSTest alias (synonym to DotNetCoreVSTest) (@augustoproiete)
- [__#3549__](https://github.com/cake-build/cake/issues/3549) Add DotNetTest alias (synonym to DotNetCoreTest) (@augustoproiete)
- [__#3548__](https://github.com/cake-build/cake/issues/3548) Add DotNetBuildServer alias (synonym to DotNetCoreBuildServer) (@augustoproiete)
- [__#3547__](https://github.com/cake-build/cake/issues/3547) Add DotNetBuild alias (synonym to DotNetCoreBuild) (@augustoproiete)
- [__#3546__](https://github.com/cake-build/cake/issues/3546) Add DotNetRestore alias (synonym to DotNetCoreRestore) (@augustoproiete)
- [__#3545__](https://github.com/cake-build/cake/issues/3545) Add DotNetClean alias (synonym to DotNetCoreClean) (@augustoproiete)
- [__#3544__](https://github.com/cake-build/cake/issues/3544) Add DotNetExecute alias (synonym to DotNetCoreExecute) (@augustoproiete)
- [__#3543__](https://github.com/cake-build/cake/issues/3543) Add DotNetRun alias (synonym to DotNetCoreRun) (@augustoproiete)
- [__#3542__](https://github.com/cake-build/cake/issues/3542) Add DotNetTool alias (synonym to DotNetCoreTool) (@augustoproiete)
- [__#3523__](https://github.com/cake-build/cake/issues/3523) Add DotNetMSBuild alias (synonym to DotNetCoreMSBuild) (@augustoproiete)
- [__#3215__](https://github.com/cake-build/cake/issues/3215) Add RunCommand with postAction parameter to DotNetCoreTool (@jenscski)
- [__#3075__](https://github.com/cake-build/cake/issues/3075) Make FilePath and DirectoryPath comparable by value (@franciscomoloureiro)
- [__#2571__](https://github.com/cake-build/cake/issues/2571) OctopusDeploy DeployTo property to take collection of string to specify multiple environments (@nckl)
- [__#2075__](https://github.com/cake-build/cake/issues/2075) Add overloads for DotNetCore*() methods taking FilePath instead of string (@dharillo)
- [__#1794__](https://github.com/cake-build/cake/issues/1794) Private is missing from ProjectReference (@franciscomoloureiro)
- [__#1616__](https://github.com/cake-build/cake/issues/1616) Error message on circular references leads to poor developer experience (@norpache)

__Bug Fixes__

- [__#1669__](https://github.com/cake-build/cake/issues/1669) Release notes does not tolerate prerelease versions (@sgronlund, @ManasviGoyal)
- [__#2443__](https://github.com/cake-build/cake/issues/2443) Erroneous "Target path must be an absolute path" when preserveFolderStructure is used with CopyFiles (@ToMee99)

__Documentation__

- [__#3602__](https://github.com/cake-build/cake/issues/3602) Switch to Cake.Tool as primary package in REAME.md (@devlead)

Pull requests were reviewed by: @augustoproiete, @devlead, @nils-a, @pascalberger, @Roemer

Fixes issues reported by: @augustoproiete, @BlythMeister, @ChrisMaddock, @devlead, @ghuntley, @gitfool, @jnm2, @luisgoncalves, @mandalorianbob, @mynkow, @nckl, @nils-a, @pascalberger, @scemino

<hr /><em>This discussion was created from the release <a href='https://github.com/cake-build/cake/releases/tag/v2.0.0-rc0001'>v2.0.0-rc0001</a>.</em>

**Comment by gitfool** [comment]
One thing that caught / confused me is that [DotNetBuildSettings.MSBuildSettings](https://github.com/cake-build/cake/blob/108d3abfd1d046a81544a039904a91fb1dc90c4d/src/Cake.Common/Tools/DotNet/Build/DotNetBuildSettings.cs#L82) is of type [DotNetCoreMSBuildSettings](https://github.com/cake-build/cake/blob/108d3abfd1d046a81544a039904a91fb1dc90c4d/src/Cake.Common/Tools/DotNetCore/MSBuild/DotNetCoreMSBuildSettings.cs#L12) instead of [DotNetMSBuildSettings](https://github.com/cake-build/cake/blob/108d3abfd1d046a81544a039904a91fb1dc90c4d/src/Cake.Common/Tools/DotNet/MSBuild/DotNetMSBuildSettings.cs#L17). Also for [DotNetCleanSettings](https://github.com/cake-build/cake/blob/108d3abfd1d046a81544a039904a91fb1dc90c4d/src/Cake.Common/Tools/DotNet/Clean/DotNetCleanSettings.cs#L48), [DotNetPackSettings](https://github.com/cake-build/cake/blob/108d3abfd1d046a81544a039904a91fb1dc90c4d/src/Cake.Common/Tools/DotNet/Pack/DotNetPackSettings.cs#L104), [DotNetPublishSettings](https://github.com/cake-build/cake/blob/108d3abfd1d046a81544a039904a91fb1dc90c4d/src/Cake.Common/Tools/DotNet/Publish/DotNetPublishSettings.cs#L178), and [DotNetRestoreSettings](https://github.com/cake-build/cake/blob/108d3abfd1d046a81544a039904a91fb1dc90c4d/src/Cake.Common/Tools/DotNet/Restore/DotNetRestoreSettings.cs#L116), so presumably this is intentional?

**Reply by augustoproiete** [reply]
Not intentional 😞 ... But thanks for catching it before we ship v2.0.0 RTM! 🙏🏻 
Tracking via https://github.com/cake-build/cake/issues/3683

**Reply by augustoproiete** [reply]
PR https://github.com/cake-build/cake/pull/3684 I'd add you as reviewer but GitHub won't let me unless you add a comment to the PR 🤷🏻‍♂️
